runtime/src/drivers.rs

```diff
@@ -446,13 +445,12 @@ impl Drivers {
         for measurement_log_entry in measurement_log.iter().take(num_measurements) {
             // Check that adding this measurement to DPE doesn't cause
             // the PL0 context threshold to be exceeded.
-            let flags = pdata.manifest1.header.flags;
-            Self::is_dpe_context_threshold_exceeded(
-                pl0_pauser_locality,
-                flags,
+            //
+            // Use the helper method here because the DPE instance holds a mutable reference to driver
+            Self::is_dpe_context_threshold_exceeded_helper(
                 pl0_pauser_locality,
+                &privilege_level,
```

Comment:
nit: Since the enum is small, can probably just pass by value instead of borrowing

Reply:
I had to be careful to instantiate the enum before a mutable reference to the `driver` variable is grabbed. In this case moving the enum would make the object invalid in the next loop iteration.

Let me know if you have any concerns about this. A deeper refactor or an alternative approach could be cleaner.

Reply:
If you derive the Copy trait on the enum does that solve the problem?

Reply:
I can't remember if you also need to derive Clone or not

Reply:
I've added a clone and now the enum is passed by value